src/Summoner/Question.hs

```diff
@@ -45,15 +46,18 @@ printQuestion question (def:rest) = do
     putTextLn $ "/" <> restSlash
 printQuestion question [] = T.putStrLn question
 
-choose :: Text -> [Text] -> IO Text
+choose :: forall a. (Show a, Read a) => Text -> [a] -> IO a
 choose question choices = do
-    printQuestion question choices
+    let showChoices = map show choices
+    printQuestion question showChoices
     answer <- prompt
-    if | T.null answer -> pure (Unsafe.head choices)
-       | answer `elem` choices -> pure answer
-       | otherwise -> do
-           errorMessage "This wasn't a valid choice."
-           choose question choices
+    if T.null answer
+        then pure (Unsafe.head choices)
+        else case readMaybe @a (T.unpack answer) of
+                Just a  -> pure a
+                Nothing -> do
```

Comment:
Opps, sorry. Now I see that this is exactly `whenNothing` function

Reply:
We have found it!